src/ApiIncrementalChecker.ts

```diff
@@ -108,6 +110,8 @@ export class ApiIncrementalChecker implements IncrementalCheckerInterface {
     this.lastUpdatedFiles = diagnostics.updatedFiles;
     this.lastRemovedFiles = diagnostics.removedFiles;
 
+    this.emitFiles();
```

Comment:
We don't seem to be testing with `if (this.enableEmitFiles) {` here?

Reply:
For `ApiIncrementalChecker` I tried to use this flag here but it does not seem to work - it keeps emitting the files even when the flag is `false` as I explained on the Technical Description of the Pull Request above. I do not know exactly why because I do not have the global understanding of the way your plugin was written, do you have any clue on this?

I verified that, for `ApiIncrementalChecker`, if I do this check with the flag `this.enableEmitFiles` inside the method `writeFile()` of `CompilerHost` it does work, that is why it is done this way at this moment.